Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewScope.swift

```diff
@@ -244,6 +247,22 @@ internal class RUMViewScope: RUMScope, RUMContextProvider {
         )
     }
 
+    private func addDiscreteUserAction(on command: RUMAddUserActionCommand) {
+        userActionScope = createDiscreteUserActionScope(on: command)
+    }
+
+    private func sendDiscreteCustomUserAction(on command: RUMAddUserActionCommand) {
+        let customActionScope = createDiscreteUserActionScope(on: command)
+        _ = customActionScope.process(
```

Comment:
This is hacky solution  but super simple 👍  - we create an unmanaged scope just for the sake of sending it a known command because we know what will happen internally (and if the internals change, this may break). It is also the very first place where we create a `RUMCommand` internally in scopes hierarchy - all other use cases that I know base on the command being consistently delivered from `RUMMonitor`. OK to me for now 👌 but let's keep an eye on this so it doesn't leak in the future.

More consistent approach would be to pin this scope on `self.customActionScope` and manage it in `self.process()`. That would require changing the `RUMUserActionScope` implementation so it can expire with `0ns` duration.

Reply:
Indeed it is a hack with minimal impact. Also it is possible to send build `ActionEvent` directly from `ViewScope` and send it to `writer`, but it is also a leak/duplication of the logic from `ActionScope` which is not nice.

Do you suggest to implement pinning to the `self.customActionScope` as part of this PR?

Reply:
Not at all, as you say this is minimal impact - good to go IMO. We may want to use sth more complex only when we have a good reason.